lib/Dialect/FIRRTL/Transforms/LowerTypes.cpp

```diff
@@ -920,16 +932,35 @@ void TypeLoweringVisitor::visitDecl(InstanceOp op) {
 }
 
 void TypeLoweringVisitor::visitExpr(SubaccessOp op) {
-  // Reads.  All writes have been eliminated before now
-
   auto input = op.input();
   auto vType = input.getType().cast<FVectorType>();
+
+  // Check for empty vectors
+  if (vType.getNumElements() == 0) {
+    Value inv = builder->create<InvalidValueOp>(vType.getElementType());
+    op.replaceAllUsesWith(inv);
+    opsToRemove.push_back(op);
+    return;
+  }
```

Comment:
It is my understanding that this is the correct behavior as defined by what the SFC's [`ZeroLengthVecs`](https://github.com/chipsalliance/firrtl/blob/master/src/main/scala/firrtl/passes/ZeroLengthVecs.scala) does.